hack/terraform/eks/main.tf

```diff
@@ -0,0 +1,126 @@
+}
+
+provider "null" {
+  version = "~> 2.1"
+}
+
+provider "template" {
+  version = "~> 2.1"
+}
+
+data "aws_eks_cluster" "cluster" {
+  name = module.eks.cluster_id
+}
+
+data "aws_eks_cluster_auth" "cluster" {
+  name = module.eks.cluster_id
+}
+
+provider "kubernetes" {
+  load_config_file       = false
+  host                   = data.aws_eks_cluster.cluster.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+  token                  = data.aws_eks_cluster_auth.cluster.token
+  # eks module not yet compatible with newer kubenetes provider 1.11.
+  version                = "1.10"
+}
+
+data "aws_availability_zones" "available" {
+}
+
+locals {
+  cluster_name = "test-eks-${random_string.suffix.result}"
+}
+
+resource "random_string" "suffix" {
+  length  = 8
+  special = false
+}
+
+resource "aws_security_group" "all_worker_mgmt" {
+  name_prefix = "all_worker_management"
+  vpc_id      = module.vpc.vpc_id
+
+  ingress {
+    from_port = var.eks_open_port_begin
+    to_port   = var.eks_open_port_end
+    protocol  = "tcp"
+
+    cidr_blocks = [
+      "0.0.0.0/0",
+      "172.16.0.0/12",
+      "192.168.0.0/16",
+    ]
+  }
+}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "2.21.0"
+
+  name                 = "test-vpc"
+  cidr                 = "10.0.0.0/16"
+  azs                  = data.aws_availability_zones.available.names
+  public_subnets       = ["10.0.4.0/24", "10.0.5.0/24", "10.0.6.0/24"]
+  enable_nat_gateway   = false
+  enable_dns_hostnames = true
+
+  tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+  }
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                      = "1"
+  }
+}
+
+module "eks" {
+  source       = "terraform-aws-modules/eks/aws"
+  cluster_name = local.cluster_name
+  subnets      = module.vpc.public_subnets
+
+  tags = {
+    Environment = "test"
+    GithubRepo  = "terraform-aws-eks"
+    GithubOrg   = "terraform-aws-modules"
+  }
+
+  vpc_id = module.vpc.vpc_id
+
+  worker_groups = [
+    {
+      name                          = "worker-group"
+      instance_type                 = "t2.medium"
+      additional_userdata           = "echo foo bar"
+      asg_desired_capacity          = var.eks_worker_count
+      public_ip = true
+      iam_instance_profile_name     = var.eks_iam_instance_profile_name
+      key_name = var.eks_key_pair_name
+    },
+  ]
+
+  worker_additional_security_group_ids = [aws_security_group.all_worker_mgmt.id]
+  map_roles                            = var.map_roles
+  map_users                            = var.map_users
+  map_accounts                         = var.map_accounts
+  manage_cluster_iam_resources         = false
+  cluster_iam_role_name                = var.eks_cluster_iam_role_name
+  manage_worker_iam_resources          = false
+  manage_worker_autoscaling_policy     = false
```

Comment:
`manage_worker_autoscaling_policy` is no longer available in terraform-aws-modules/eks/aws 9.0.0 (https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/9.0.0?tab=inputs), which was installed by default when I ran your `eks` script. I would suggest to remove it or find the new equivalent. Maybe it would also be worth it to refine the version requirement above, by auditing the attributes you are using:
```
provider "aws" {
  version = ">= 2.28.1"
  region  = var.region
}
```

Reply:
ok will do.

Reply:
removed manage_worker_autoscaling_policy